app/src/eds/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java

```diff
@@ -0,0 +1,116 @@
+ *  (at your option) any later version.
+ *
+ *  Health Network QIS App is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.eyeseetea.malariacare.strategies;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.eyeseetea.malariacare.DashboardActivity;
+import org.eyeseetea.malariacare.LoginActivity;
+import org.eyeseetea.malariacare.ProgressActivity;
+import org.eyeseetea.malariacare.R;
+import org.eyeseetea.malariacare.data.database.model.User;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.usecase.LoadUserAndCredentialsUseCase;
+import org.eyeseetea.malariacare.domain.usecase.LoginUseCase;
+import org.hisp.dhis.client.sdk.ui.views.FontButton;
+
+public class LoginActivityStrategy extends ALoginActivityStrategy {
+
+    private static final String TAG = ".LoginActivityStrategy";
+
+    public LoginActivityStrategy(LoginActivity loginActivity) {
+        super(loginActivity);
+    }
+
+    @Override
+    public void onCreate() {
+        if (existsLoggedUser()) {
+            LoadUserAndCredentialsUseCase loadUserAndCredentialsUseCase =
+                    new LoadUserAndCredentialsUseCase(loginActivity);
+
+            loadUserAndCredentialsUseCase.execute();
+
+            finishAndGo(DashboardActivity.class);
+        } else {
+            loginActivity.runOnUiThread(new Runnable() {
+                public void run() {
+                    addDemoButton();
+                }
+            });
+        }
+    }
+
+
+    private boolean existsLoggedUser() {
+        return User.getLoggedUser() != null && !ProgressActivity.PULL_CANCEL;
+    }
+
+    private void addDemoButton() {
+        ViewGroup loginViewsContainer = (ViewGroup) loginActivity.findViewById(
+                R.id.layout_login_views);
+
+        loginActivity.getLayoutInflater().inflate(R.layout.demo_login_button, loginViewsContainer,
+                true);
+
+        FontButton demoButton = (FontButton) loginActivity.findViewById(R.id.demo_login_button);
+
+        demoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                Credentials demoCrededentials = Credentials.createDemoCredentials();
+
+                loginActivity.mLoginUseCase.execute(demoCrededentials,
+                        new LoginUseCase.Callback() {
+                            @Override
+                            public void onLoginSuccess() {
+                                loginActivity.onSuccess();
```

Comment:
@idelcano I think that after demo login we should not navigate to ProgressActvity, we should execute demo pull as in pictureapp